clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -924,6 +935,15 @@ void CodeGenModule::Release() {
                                            RD->getAttr<SYCLUsesAspectsAttr>()));
       }
     }
+
+    // Emit metadata for all aspects defined in the aspects enum.
+    if (AspectsEnumDecl) {
+      llvm::NamedMDNode *AspectEnumValsMD =
+          TheModule.getOrInsertNamedMetadata("intel_sycl_aspects");
```

Comment:
The same could be said for every other LLVM attribute described in this design (e.g. `intel_declared_aspects`, `intel_used_aspects`).  We have existing LLVM attributes named this way too (e.g. `intel_reqd_sub_group_size`).  If we think we want to rename SYCL-related LLVM attributes to make them easier to upstream, we should make some sort of global decision and rename them all.

Reply:
`intel_reqd_sub_group_size` was introduced for Intel specific extension, where as aspect is already SYCL core feature.

Reply:
That's right, but what "intel_" in "intel_sycl_aspects" refers to? Is this the name of some type in the extension? I assumed that it refers to "Intel specific implementation" and this will be difficult to upstream.
I think we agree that things should be named after the spec entities, rather than implementation names like dpcpp, intel, etc. if we aim to commit this code to LLVM project.

Reply:
I'm not opposed to renaming these LLVM IR attributes, I just think they should all be named consistently.  If we want to remove the "intel" prefix, maybe the  new names could be like:

* `intel_types_that_use_aspects` -> `sycl_types_that_use_aspects`
* `intel_declared_aspects` -> `sycl_declared_aspects`
* `intel_used_aspects` -> `sycl_used_aspects`
* `intel_fixed_targets` -> `sycl_fixed_targets`
* `intel_sycl_aspects` -> `sycl_aspects`

In addition, we should identify other sycl-related LLVM IR attributes that start with "intel" and rename them to start with "sycl".  This PR could rename the attributes defined in the OptionalDeviceFeratures design, and we could create a tracker to capture the work required to rename any other attributes.

Alternatively, we could keep the names as "intel" in this PR and create a new PR that renames everything to "sycl" prefix.

Reply:
Yes, please, update other LLVM IR attributes (if it's possible to do w/o ABI breaking) too. This PR mentions only one attribute and I was not aware that there other similar attributes.

Reply:
I am happy to go rename the metadata nodes related to OptionalDeviceFeratures, but I would prefer to do it in a separate PR to avoid polluting this one. That said, since `intel_sycl_aspects` is introduced with this I will make the renaming of it as part of this PR.